Webpack url not found

from @msebas regarding #6 

After building your fix `5d3ec27` I installed it to the test project. Trying to run the dev server using `npm run serve` I got an unrelated error stating that `Module not found: Error: Can't resolve 'url' in ...`, so I installed url using `npm install --save-dev url` and checked if `node_modules/@pyodide/webpack-plugin` was not updated by npm. Now the dev server starts up.
To do something with pyodide I updated `src/main.js` to be
```
import { createApp } from 'vue'
import App from './App.vue'
import {loadPyodide} from "pyodide";

createApp(App).mount('#app')
let pyodide = null;
async function main() {
  pyodide = await loadPyodide({ indexURL: `${window.location.origin}/pyodide` });
  return pyodide.runPython(`
    import sys
    sys.version
  `)
}
main().then(r => console.log(r));
```
(following the example in the `README.md`)

The dev-Server starts now, but loading `http://localhost:8080/` leads to a console error
```
Uncaught (in promise) Error: Cannot find module 'http://localhost:8080/pyodide/pyodide.asm.js'
    webpackEmptyAsyncContext app.js:99
    promise callback*webpackEmptyAsyncContext app.js:98
    loadScript compat.ts:158
    loadPyodide pyodide.ts:308
    main main.js:8
    <anonymous> main.js:15
    js app.js:63
    __webpack_require__ app.js:277
    __webpack_exports__ app.js:1499
    O app.js:326
    <anonymous> app.js:1500
    <anonymous> app.js:1502
app.js:99:11
```
which is cause by the function 
```
function webpackEmptyAsyncContext(req) {
	// Here Promise.resolve().then() is used instead of new Promise() to prevent
	// uncaught exception popping up in devtools
	return Promise.resolve().then(function() {
		var e = new Error("Cannot find module '" + req + "'");
		e.code = 'MODULE_NOT_FOUND';
		throw e;
	});
}
```
that seems to be loaded by pyodide using
`loadScript = async url => await __webpack_require__("./node_modules/pyodide lazy recursive")(url)`
where `url` is the string `"http://localhost:8080/pyodide/pyodide.asm.js"` (strangely calling the URL itself returns said script from the pyodide main module). 

Guessing that the value of `url` might be not valid for webpacks resolving mechanism I dropped the `indexURL` from `main.js` (loading pyodide via `pyodide = await loadPyodide();`). This resulted in the same error, only the value of `url` changed to `"webpack-internal:///./node_modules/pyodide/pyodide.asm.js"`.

I am a little bit at a loss what makes webpack screw up here, because the first value of `url` serves what is required.

--- a/README.md
+++ b/README.md
@@ -79,6 +79,45 @@ Default: `https://cdn.jsdelivr.net/pyodide/v${installedPyodideVersion}/full/`\
 Required: false\
 _Description_: CDN endpoint for python packages. This option differs from [loadPyodide indexUrl](https://pyodide.org/en/stable/usage/api/js-api.html) in that it only impacts pip packages and _does not_ affect the location the main pyodide runtime location. Set this value to "" if you want to keep the pyodide default of accepting the indexUrl.
 
+## Known issues with esm
+
+Depending on your webpack configuration you may run into issues with webpack trying to parse your pyodide.mjs file.
+
+**Can't resolve 'url' in ...**
+
+[Issue #8](https://github.com/pyodide/pyodide-webpack-plugin/issues/8) deals with this error and pyodide esm. To fix this issue:
+
+- `npm i -D url`
+- Add a [fall back](https://webpack.js.org/configuration/resolve/#resolvefallback) to your webpack config
+  ```js
+  resolve: {
+    fallback: {
+      url: require.resolve("url/"),
+    },
+  },
+  ```
+
+**Cannot find module '<...>/pyodide.asm.js.**
+
+This can happen when webpack munges esm import statements to \_\_webpack_require\_\_ when you actually intend import to work in the browser.
+
+- `npm i -D string-replace-loader`
+- Add a [rule](https://webpack.js.org/configuration/module/#rule) to your webpack config
+  ```js
+  {
+    test: /pyodide\.m?js$/,
+    use: [
+      {
+        loader: 'string-replace-loader',
+        options: {
+          search: 'import(',
+          replace: 'import(/* webpackIgnore: true */ '
+        }
+      }
+    ]
+  }
+  ```
+
 ## Contributing
 
 Please view the [contributing guide](./CONTRIBUTING.md) for tips on filing issues, making changes, and submitting pull requests. Pyodide is an independent and community-driven open-source project. The decision-making process is outlined in the [Project governance](https://pyodide.org/en/stable/project/governance.html).

